vortex-scalar/src/utf8.rs

```diff
@@ -21,8 +21,19 @@ impl<'a> Utf8Scalar<'a> {
         self.value.as_ref().cloned()
     }
 
-    pub fn cast(&self, _dtype: &DType) -> VortexResult<Scalar> {
-        todo!()
+    pub(crate) fn cast(&self, dtype: &DType) -> VortexResult<Scalar> {
```

Comment:
Can you cast a string into a number if it parses correctly?

Reply:
ditto for binary?

Reply:
Parsing a string seems wrong to me because it's so much more work. Casting from string to binary (i.e. forgetting the UTF8-ness) seems correct to me. I guess Binary to any integer type should be fine, erroring if the binary is not the right length. Particularly because if we add FixedWidthBinary, I would think `binary -> fixedwidthbinary(4) -> u32` should work.

Should I add the String to Binary case to this PR?

Reply:
I think it's okay to add a FLUP issue with various casts we want to support (e.g., bool to int, utf8 to binary, etc)

Reply:
https://github.com/spiraldb/vortex/issues/2031